<launch>
    <!-- Model Type -->
    <!-- <arg name="model_type" default="resnet50dilated+ppm_deepsup" /> -->
    <arg name="model_type" default="hrnetv2+c1" />
    <!-- <arg name="depth_model_type" default="lite_mono" /> -->
    <arg name="depth_model_type" default="distdepth" />

    <!-- Topics -->
    <arg name="pointcloud_topic" default="/semantic_pointcloud" />
    <arg name="seg_rgb_topic" default="/tesse/seg_cam/rgb/image_raw" />

    <arg name="rgb_topic" default="/tesse/left_cam/rgb/image_raw" />
    <arg name="rgb_info_topic" default="/tesse/left_cam/camera_info" />

    <arg name="depth_topic" default="/tesse/depth_cam/mono/image_raw" />
    <arg name="depth_info_topic" default="/tesse/depth_cam/mono/camera_info" />

    <arg name="odom_publish_topic" default="/tesse/odom" />

    <!-- Input Topics -->
    <!-- <arg name="image_topic" default="/camera/color/image_raw" /> -->
    <arg name="image_topic" default="/camera_array/cam0/image_raw" /> 
    <!-- <arg name="image_topic" default="/camera/image_raw" /> -->


    <!-- <arg name="odometry_topic" default="/vins_estimator/odometry" /> -->
    <arg name="odometry_topic" default="/aft_mapped_to_init" />
    <!-- <arg name="odometry_topic" default="//Odometry" /> -->

    <arg name="image_info_topic" default="/camera/camera_info" />
    <arg name="depth_image_topic" default="camera/aligned_depth_to_color/image_raw" />

    <arg name="bow_vector" default="/bow_query" />
    <arg name="bow_vector_pub_topic" default="/kimera_vio_ros/bow_query" />


    <!-- Semantic Segmentation Publisher Node -->
    <node name="semantic_segmentation_publisher_node" pkg="semantic_segmentation_publisher" type="semantic_segmentation.py" output="screen">

        <param name="model_type" value="$(arg model_type)" />
        <param name="depth_model_type" value="$(arg depth_model_type)" />

        <param name="pointcloud_topic" value="$(arg pointcloud_topic)" />

        <param name="rgb_topic" value="$(arg rgb_topic)" />
        <param name="rgb_info_topic" value="$(arg rgb_info_topic)" />
        <param name="depth_image_topic" value="$(arg depth_image_topic)" />
        <param name="depth_info_topic" value="$(arg depth_info_topic)" />

        <param name="seg_rgb_topic" value="$(arg seg_rgb_topic)" />

        <param name="depth_topic" value="$(arg depth_topic)" />
        <param name="image_topic" value="$(arg image_topic)" />
        <param name="image_info_topic" value="$(arg image_info_topic)" />

        <param name="odometry_topic" value="$(arg odometry_topic)" />
        <param name="odom_publish_topic" value="$(arg odom_publish_topic)" />


        <param name="bow_vector" value="$(arg bow_vector)" />

        <param name="bow_vector_pub_topic" value="$(arg bow_vector_pub_topic)" />


    </node>

    <!-- RViz Node -->
    <node type="rviz" name="rviz_semantic_segmentation" pkg="rviz" args="-d $(find semantic_segmentation_publisher)/rviz/tesse_rviz.rviz" />

    <!-- Nodelet Manager -->
    <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager"/>

    <!-- Depth Image Registration Nodelet -->
    <!-- <node pkg="nodelet" type="nodelet" name="depth_image_registration" args="load depth_image_proc/register nodelet_manager" output="screen">
        <param name="queue_size" value="20"/>

        <remap from="rgb/camera_info" to="$(arg rgb_info_topic)"/>
        <remap from="depth/camera_info" to="$(arg depth_info_topic)"/>
        <remap from="depth/image_rect" to="$(arg depth_topic)"/>
        <remap from="depth_registered/camera_info" to="/registered_depth_info_topic"/>
        <remap from="depth_registered/image_rect" to="/registered_depth_image_topic"/>
    </node> -->

    <!-- Point Cloud Nodelet -->
    <node pkg="nodelet" type="nodelet" name="cloudify" args="load depth_image_proc/point_cloud_xyzrgb nodelet_manager -no-bond" output="screen">
        <param name="queue_size" value="20"/>

        <remap from="rgb/camera_info" to="$(arg rgb_info_topic)"/>
        <remap from="rgb/image_rect_color" to="$(arg seg_rgb_topic)"/>
        <remap from="depth_registered/image_rect" to="$(arg depth_topic)"/>
        <remap from="depth_registered/points" to="$(arg pointcloud_topic)"/>
    </node>
    <!-- <node pkg="tf" type="static_transform_publisher" name="base_link_gt_to_left_cam"
            args="0 0 0 0 0 0 -1 base_link_get left_cam 100"/> 

    <node pkg="tf" type="static_transform_publisher2" name="base_to_left_cam"
            args="0 0 0 0 0 0 -1 world left_cam 100"/> -->

</launch>
 